test/discrepancies/Nonce.js

```diff
@@ -360,14 +331,357 @@ describe('@discrepancies - Nonce Test Suite', async () => {
+  })
+
+  it('should update nonce after successful ERC721 token call', async function () {
+    erc721Contract = await Utils.deployERC721Contract();
+    await Utils.updateTokenKeysViaHapi(tokenAddress, [
+      await tokenCreateContract.getAddress(),
+      await tokenTransferContract.getAddress(),
+    ]);
+
+    const snBefore = await getServicesNonce(signers[0].address);
+    const mnBefore = await getMirrorNodeNonce(signers[0].address);
+
+    mintedTokenSerialNumber = await Utils.mintNFT(
+        tokenCreateContract,
+        tokenAddress
+    );
+
+    const snAfter = await getServicesNonce(signers[0].address);
+    const mnAfter = await getMirrorNodeNonce(signers[0].address);
+
+    expectIncrementedNonce(snBefore, mnBefore, snAfter, mnAfter);
+  });
+
+  it('should update nonce after successful call to Ethereum Precompiles', async function () {
+    const contractFactory = await ethers.getContractFactory(Constants.Contract.EthNativePrecompileCaller);
+    const contract = await contractFactory.deploy({
+      gasLimit: 15_000_000,
+    });
+
+    const UNSIGNED_DATA = 'Hello Eth Native Precompiles!';
+    let signer = (await ethers.getSigners())[0];
+    let hashedData = ethers.hashMessage(UNSIGNED_DATA);
+    let signedData = await signer.signMessage(UNSIGNED_DATA);
+    let signerAddr = signer.address.toLowerCase().replace('0x', '');
+
+    const splitSignature = ethers.Signature.from(signedData);
+
+    let v = splitSignature.v;
+    let r = splitSignature.r;
+    let s = splitSignature.s;
+
+    const callData = `0x${TestUtils.to32ByteString(hashedData)}${TestUtils.to32ByteString(v)}${TestUtils.to32ByteString(r)}${TestUtils.to32ByteString(s)}`;
+
+    const snBefore = await getServicesNonce(signers[0].address);
+    const mnBefore = await getMirrorNodeNonce(signers[0].address);
+
+    const result = await contract.call0x01(callData);
+    const rec = await result.wait();
+    expect(rec.logs[0].data).to.contain(signerAddr);
+
+    const snAfter = await getServicesNonce(signers[0].address);
+    const mnAfter = await getMirrorNodeNonce(signers[0].address);
+
+    expectIncrementedNonce(snBefore, mnBefore, snAfter, mnAfter);
+  });
+
+  it('should update nonce after unsuccessful contract deploy with CREATE2 ', async function () {
+    const firstTx = await internalCalleeContract.deployViaCreate2(1);
+    await firstTx.wait();
+
+    const snBefore = await getServicesNonce(signers[0].address);
+    const mnBefore = await getMirrorNodeNonce(signers[0].address);
+
+    await Utils.expectToFail(
+        internalCalleeContract.deployViaCreate2(1, {gasLimit: 500000})
+    );
+
+    const snAfter = await getServicesNonce(signers[0].address);
+    const mnAfter = await getMirrorNodeNonce(signers[0].address);
+
+    expectIncrementedNonce(snBefore, mnBefore, snAfter, mnAfter);
+  });
+
+  it('should not update nonces when deploying on an address with an already existing account', async function () {
```

Comment:
nit: test name is a little misleading. I assumed there was a contract deployment but rather it's a hollow account -> full account -> deleted account -> hollow account flow.
The nonces are updated when you execute a transfer. In the other 2 checks you confirm the hollow account nonce is zero.
Just update the test description so it's clear

Reply:
Done.